projects/plugins/jetpack/tests/php/json-api/test-class.wpcom-json-api-site-settings-v1-4-endpoint.php

```diff
@@ -0,0 +1,286 @@
+		$endpoint->api->method       = 'POST';
+
+		return $endpoint->callback( '', $blog_id );
+	}
+
+	/**
+	 * Data provider that contains keys we expect to see returned by the settings endpoint and their default value.
+	 *
+	 * @return array[ $setting_name, $expected_default_value ]
+	 */
+	public function setting_default_key_values() {
+		return array(
+			'woocommerce_store_address'      => array( 'woocommerce_store_address', '' ),
+			'woocommerce_store_address_2'    => array( 'woocommerce_store_address_2', '' ),
+			'woocommerce_store_city'         => array( 'woocommerce_store_city', '' ),
+			'woocommerce_default_country'    => array( 'woocommerce_default_country', '' ),
+			'woocommerce_store_postcode'     => array( 'woocommerce_store_postcode', '' ),
+			'woocommerce_onboarding_profile' => array( 'woocommerce_onboarding_profile', array( 0 => false ) ),
+		);
+	}
+
+	/**
+	 * Data provider to test setting value pairs in GET request.
+	 *
+	 * @return array[ $setting_name, $setting_value ]
+	 */
+	public function setting_value_pairs_get_request() {
+		return array(
+			'woocommerce_store_address'      => array( 'woocommerce_store_address', 'Street 34th 1/2' ),
+			'woocommerce_store_address_2'    => array( 'woocommerce_store_address_2', 'Apt #1' ),
+			'woocommerce_store_city'         => array( 'woocommerce_store_city', 'City' ),
+			'woocommerce_default_country'    => array( 'woocommerce_default_country', 'US:NY' ),
+			'woocommerce_store_postcode'     => array( 'woocommerce_store_postcode', '98738' ),
+			'woocommerce_onboarding_profile' => array( 'woocommerce_onboarding_profile', array( 'test' => 'test value' ) ),
+		);
+	}
+
+	/**
+	 * Data provider to test setting value pairs in POST request.
+	 *
+	 * @return array[ $setting_name, $setting_value, $expected_value ]
+	 */
+	public function setting_value_pairs_post_request() {
+		return array(
+			'woocommerce_store_address'                 => array( 'woocommerce_store_address', '<h1>Street 34th 1/2</h1>', 'Street 34th 1/2' ),
+			'woocommerce_store_address_2'               => array( 'woocommerce_store_address_2', '<h2>Apt #1</h2>', 'Apt #1' ),
+			'woocommerce_store_city'                    => array( 'woocommerce_store_city', '<h3>City</h3>', 'City' ),
+			'woocommerce_default_country'               => array( 'woocommerce_default_country', '<p>US:NY</p>', 'US:NY' ),
+			'woocommerce_store_postcode'                => array( 'woocommerce_store_postcode', '<div>98738</div>', '98738' ),
+			'woocommerce_store_postcode script tag'     => array( 'woocommerce_store_postcode', '<script>98738</script>', '' ),
+			'woocommerce_onboarding_profile'            => array( 'woocommerce_onboarding_profile', array( 'test_key' => '<strong>test value</strong>' ), array( 'test_key' => 'test value' ) ),
+			'woocommerce_onboarding_profile script tag' => array( 'woocommerce_onboarding_profile', array( 'test_key' => '<script>test value</script>' ), array( 'test_key' => '' ) ),
```

Comment:
Could we add a case here that passes a non-array value and checks if we get an array back?

Reply:
Good call. I also went ahead and added a unit test using actual sample data and discovered an issue with boolean values and sub-arrays. These issues should be resolved now.